src/variety/pencils.js

```diff
@@ -0,0 +1,896 @@
+	mouseinput_number : function(){
+		if(this.mousestart){ this.inputqnum_pencils();}
+	},
+	mouseinput : function(){ // オーバーライド
+		if(this.inputMode==='undef'){
+			if(this.mousestart){ this.inputqnum_pencils();}
+		}
+		else{ this.common.mouseinput.call(this);}
+	},
+	mouseinput_auto : function(){
+		if(this.puzzle.playmode){
+			if(this.mousestart || this.mousemove){
+				if(this.btn==='left'){
+					if(this.isBorderMode()) {
+						this.inputborder();
+					} else{
+						this.inputLineOrArrow();
+					}
+				} else {
+					this.inputpeke();
+				}
+			}
+			else if(this.mouseend && this.notInputted() && !this.isBorderMode()) {
+				this.inputBGcolor();
+			}
+		}
+		else if(this.puzzle.editmode){
+			if(this.mousestart || this.mousemove){ this.inputarrow_cell();}
+			else if(this.mouseend && this.notInputted()){ this.inputqnum_pencils();}
+		}
+	},
+
+	inputLineOrArrow : function(){
+		var pos, border;
+		pos = this.getpos(0);
+		if(this.prevPos.equals(pos)){ return;}
+		border = this.prevPos.getnb(pos);
+		var cell0 = border.sidecell[0], cell1 = border.sidecell[1];
+		
+		if(!border.isnull){
+			if(this.inputData===null) {
+				// Do not use setPencilArrow here, leave the border intact
+				if(border.isvert && (cell0.anum === border.LT || cell1.anum === border.RT)) {
+					if(cell0.anum === border.LT) {cell0.setAnum(-1);}
+					if(cell1.anum === border.RT) {cell1.setAnum(-1);}
+					this.inputData = 0;
+				}
+				if(!border.isvert && (cell0.anum === border.UP || cell1.anum === border.DN)) {
+					if(cell0.anum === border.UP) {cell0.setAnum(-1);}
+					if(cell1.anum === border.DN) {cell1.setAnum(-1);}
+					this.inputData = 0;
+				}
+			}
+
+			if(this.inputData===null){ this.inputData=(border.isLine()?0:1);}
+			if(this.inputData===1 && !border.ques){
+				if(border.qans === 1) {
+					var dir = this.getDrawArrowDirection(border);
+					if(dir > 0) {
+						cell1.setPencilArrow(border.isvert ? border.RT : border.DN, false);
+					} else {
+						cell0.setPencilArrow(border.isvert ? border.LT : border.UP, false);
+					}
+				} else {
+					border.setLine();
+				}
+			}
+			else if(this.inputData===0){ border.removeLine();}
+			border.draw();
+		}
+		this.prevPos = pos;
+	},
+
+	getDrawArrowDirection: function(border) {
+		if(border.sidecell[0].isnull) {return +1;}
+		if(border.sidecell[1].isnull) {return -1;}
+
+		if(border.sidecell[0].qnum > 0) {return +1;}
+		if(border.sidecell[1].qnum > 0) {return -1;}
+
+		if(border.sidecell[1].lcnt > 0 && border.sidecell[0].lcnt === 0) {return +1;}
+		if(border.sidecell[0].lcnt > 0 && border.sidecell[1].lcnt === 0) {return -1;}
+
+		if(border.sidecell[0].room.pencil) {return +1;}
+		if(border.sidecell[1].room.pencil) {return -1;}
+
+		var pos = this.getpos(0);
+		var dir = this.prevPos.getdir(pos,2);
+
+		return (dir === border.RT || dir === border.DN) ? +1 : -1;
+	},
+
+	mouseinput_clear : function(){
+		var cell = this.getcell();
+		var dir = cell.qdir || cell.anum;
```

Comment:
I personally find boolean "or" to choose between integer values confusing. I'd prefer `cell.qdir ? cell.qdir : cell.anum` if you want to keep it compact, and/or a method `cell.getDir()` or so that wraps this or an explicit if/else.

Reply:
I've added a new function.

Reply:
Thanks!